src/workflow/snakemake/snakemake.py

```diff
@@ -0,0 +1,190 @@
+            work_dir (str): working directory
+            **snakemake_kwargs: keyword arguments to pass to the snakemake engine
+
+        Returns:
+        - Status code of the workflow run.
+        """
+
+        # Serialize Snakefile for workflow
+        snakefile = self.serialize_workflow(benchmark, work_dir)
+
+        # Prepare the argv list
+        argv = self._prepare_argv(snakefile, cores, dryrun, work_dir, **snakemake_kwargs)
+
+        # Execute snakemake script
+        success = snakemake_cli(argv)
+
+        return success
+
+    def serialize_workflow(self, benchmark: Benchmark, output_dir: str = os.getcwd()):
+        """
+        Serializes a Snakefile for the benchmark.
+
+        Args:
+            benchmark (Benchmark): benchmark to serialize
+            output_dir (str): output directory for the Snakefile
+
+        Returns:
+        - Snakefile path.
+        """
+        os.makedirs(output_dir, exist_ok=True)
+
+        benchmark_file = benchmark.get_definition_file()
+
+        # Serialize Snakemake file
+        snakefile_path = os.path.join(output_dir, 'Snakefile')
+        with open(snakefile_path, 'w') as f:
+            self._write_snakefile_header(f)
+            self._write_includes(f, INCLUDES)
+
+            # Load benchmark from yaml file
+            f.write(f'benchmark = load("{benchmark_file}")\n\n')
+
+            # Create capture all rule
+            f.write("all_paths = sorted(benchmark.get_output_paths())\n")
+            f.write("create_all_rule(all_paths)\n\n")
+
+            # Create node rules
+            f.write("nodes = benchmark.get_nodes()\n")
+            f.write("for node in nodes:\n")
+            f.write("    create_node_rule(node, benchmark)\n\n")
+
+        return snakefile_path
+
+    def run_node_workflow(self, node: BenchmarkNode, cores: int = 1, dryrun: bool = False, work_dir: str = os.getcwd(),
+                          **snakemake_kwargs):
+        """
+        Serializes & runs benchmark node workflow using snakemake.
+
+        Args:
+            node (Benchmark): benchmark node to run
+            cores (int): number of cores to run
+            dryrun (bool): validate the snakemake workflow with the benchmark without actual execution
+            work_dir (str): working directory
+            **snakemake_kwargs: keyword arguments to pass to the snakemake engine
+
+        Returns:
+        - Status code of the workflow run.
+        """
+
+        os.makedirs(work_dir, exist_ok=True)
+
+        # Serialize Snakefile for node workflow
+        snakefile = self.serialize_node_workflow(node, work_dir)
+
+        # Prepare the argv list
+        argv = self._prepare_argv(snakefile, cores, dryrun, work_dir, **snakemake_kwargs)
+
+        # Execute snakemake script
+        success = snakemake_cli(argv)
+
+        return success
+
+    def serialize_node_workflow(self, node: BenchmarkNode, output_dir: str = os.getcwd()):
```

Comment:
[unstructured thoughts/rambling] I wonder how to align these viewpoints for ruleset rendering (benchmark vs node) as compared to the snakemake command calls themselves. So this is the most robust and user-friendly. I mean we generate the ruleset via omni_workflow, but then we execute it with a somewhat standardized version of `snakemake --cores x --use-singularity` or whatever. But we could also modify the snakemake call to `snakemake -R --until c --force --retries 200` (extreme case here). Perhaps combining snakemake command call arguments and ruleset generation patterns can provide extra usecases.